File: src/controller/dto/outgoing/chore.dto.ts
import { Transform } from 'class-transformer';
import { Subtask } from '../../../model/subtask.entity';

export class ChoreDTO {
  id!: number;
  title!: string;
  description!: string;
  @Transform(({ value }) => value.toISOString().split('T')[0])
  deadline!: Date;
  priorityLevel!: number;
  done!: boolean;
  subtasks!: Subtask[];
}

File: bnim2219-spring/src/main/java/edu/bbte/idde/bnim2219/spring/controller/dto/outgoing/ChoreDTO.java
package edu.bbte.idde.bnim2219.spring.controller.dto.outgoing;

import lombok.*;

import java.sql.Date;
import java.util.Collection;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class ChoreDTO {
    private Long id;
    private String title;
    private String description;
    private Date deadline;
    private Integer priorityLevel;
    private Boolean done;
    private Collection<SubtaskDTO> subtasks;
}
